Sources/NIOHTTPCompression/HTTPResponseDecompressor.swift

```diff
@@ -0,0 +1,203 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CNIOExtrasZlib
+import NIO
+import NIOHTTP1
+
+/// Specifies how to limit decompression inflation.
+public enum DecompressionLimit {
+    /// No limit will be set.
+    case none
+    /// Limit will be set on the request body size.
+    case size(Int)
+    /// Limit will be set on a ratio between compressed body size and decompressed result.
+    case ratio(Int)
+
+    func exceeded(compressed: Int, decompressed: Int) -> Bool {
+        switch self {
+        case .none:
+            return false
+        case .size(let allowed):
+            return compressed > allowed
+        case .ratio(let ratio):
+            return decompressed > compressed * ratio
+        }
+    }
+}
+
+public enum DecompressionError: Error {
+    case limit
+    case inflationError(Int32)
+    case initializationError(Int32)
+}
+
+public final class HTTPResponseDecompressor: ChannelDuplexHandler, RemovableChannelHandler {
```

Comment:
the name of this class needs to start with `NIO` as per https://github.com/apple/swift-nio/blob/master/docs/public-api.md

Reply:
renamed